properties.md

```diff
@@ -0,0 +1,140 @@
+
+More examples of the pattern may be found [here](https://en.wikipedia.org/wiki/Attribute%E2%80%93value_pair).
+
+### The resolver consumer
+
+One major consumer of bundle properties is OLM, which reads them to determine the surface that is exposed to the dependency resolver.
+
+The resolver and the registry apis already deal with certain aspects of a bundle via opaque `properties` via the `ListBundles` grpc API:
+
+```proto
+message Property{
+  string type = 1;
+  string value = 2;
+}
+
+message Bundle{
+    ... 
+	repeated Property properties = 12;
+    ...
+}
+```
+
+Many of the other bundle fields are replicated as properties. Here are the properties for an example bundle, rendered as psuedo-`json`:
+
+```json5
+{
+    "type":  "olm.package",
+    "value": `{"packageName":"etcd","version":"0.9.2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdCluster","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdRestore","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdBackup","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.label",
+    "value": `{"label":"testlabel"}`,
+}
+```
+
+Properties intentionally closely align with the format for [`dependencies.yaml`](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-dependencies).
+
+Today, all properties are derived from other fields in a bundle. There is no way to write a new property `type` for a consumer (say, a newer version of the resolver, or an external tool that processes sets of bundles).
+
+### Other consumers
+
+Catalog and Bundles are consumed in many other contexts than the resolver. The package-server component presents catalog content for the purpose of selecting packages and channels, for example. 
+
+### Goals
+
+- Define the format for explicitly adding properties to a bundle.
+
+## Proposal
+
+A new file is accepted in the `metadata` folder for a bundle: `properties.(yaml|json)`
+
+Here is an example of `properties.yaml`:
+
+```yaml=
+properties:
+- type: olm.package
+  value:
+    packageName: prometheus
+    version: "0.27.0"
+- type: olm.gvk
+  value:
+    group: etcd.database.coreos.com
+    kind: EtcdCluster
+    version: v1beta2
+- type: olm.label
+  value: "LTS"
+```
+
+If properties are defined, they take precendence over the values derived from the bundle contents. In the above example, the `version` of the bundle will be considered `0.27.0` regardless of what the `ClusterServiceVersion`'s `spec.version` field indicates.
+
+### Validate properties
+
+Property validation is the domain of the consumer, not the producer. 
+
+However, `opm` will ship with validation for all properties that the OLM resolver understands, and they will be validated as part of the `opm alpha bundle validate` command and the `operator-sdk` bundle validation. 
+
```

Comment:
I think we need to clarify that the property validations would be implemented in the operator-framework/api and then, would be used by the respective tools/commands.